### ISSUE TITLE
Feature request: synonyms for icon names

### ISSUE THREAD

**Issue by jakamusic** [iconify/iconify-figma#40]
## Description
It would be super helpful for the plugin to be able to find icons based on a set of synonyms.

## Example
in Fluent UI the X icon is called `dismiss` but to be able to find it also by `exit`, `times`, `close` ...

## Implementation idea
Define a set of synonyms for all icon packs to be the same and then compare search term with names and alternatives (connected words), it is a lot of effort though. 

## Possible issues
With this approach, some icons that do not fit the search terms could be displayed - but is it better to show too many sometimes with one of them being correct or to always show just the exact match?

Possible solution for this issue: a setting to search literally or expanded. 

--- 

I understand this is a BIG feature request, but it's just something I've been thinking of for a while.

**Comment by cyberalien** [comment]
Problem with that is list of synonyms needs to be created and maintained. This cannot be a generic dictionary, it needs to be specific to icons.

That is not a simple task. If there is a library or something that can be used for it, I can try integrating it in search engine, but otherwise there is very little chance of this happening due to time constraints.

**Comment by jakamusic** [comment]
Yep, though so. Yeah, the issue with a library would be that those are not necessarily synonyms per-se, more like alternative words designer use. 

I'm just thinking here, but what if we were to do a dictionary for the most used icons like: 

Trash: bin, can
Dismiss: cancel, remove, exit, times, close 
Plus: add
...

One other thing I'm thinking about is, some of the icon providers have their own websites with search, which works in a similar fashion to what I'm describing.

For example if you search for `remove` on [IonIcons](https://ionic.io/ionicons) it also shows icon called close. Same for [Hero icons](https://heroicons.com).

I'm not sure how you implement icon packs, but if that's maybe something to consider, that some of them are already doing it?

**Comment by cyberalien** [comment]
Yes, some icon sets do have that data, but it is specific to that icon set. Those keywords might not be suitable for other icons and it won't have keywords for icons not in icon set.

It probably can be used as base for dictionary, but there is still an issue of maintenance. Those lists can only be used as base, not as full dictionary because they are not complete.

**Comment by jakamusic** [comment]
Completely understand. I'll keep my eyes peeled and keep thinking of a better and more maintainable approach than the ones above.
